AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/scopedstorage/MoveFile.kt

```diff
@@ -39,56 +39,107 @@ import java.io.File
+        Timber.d("no-op - source deleted: '$sourceFile'")
+        // since the file was deleted, we can't know the size. Report 0 file size
+        context.reportProgress(0)
+    }
+
+    /**
+     * Assumes source and destination exists. Deals with this unexpected case in the following way:
+     * * if both files are the same, according to canonical path, throws a EquivalentFileException
+     * * if destination is a directory, report a a FileDirectoryConflictException
+     * * if destination is a copy of the source, delete the source, as it's copied,
+     * * if destination is a strict prefix of the source, delete the destination so that it can be copied entirely
+     * * if destination seems to be distinct from source,
+     *
+     * @throws EquivalentFileException sourceFile == destinationFile
+     * @throws MissingDirectoryException if source or destination's directory is missing
+     */
+    private fun destinationExists(context: MigrationContext) {
+        if (sourceFile.file.canonicalPath == destinationFile.canonicalPath) {
+            // Deletion is destructive if both files are the same
+            throw EquivalentFileException(sourceFile.file, destinationFile)
+        }
 
-        if (destinationExists && destinationFile.isDirectory) {
+        if (destinationFile.isDirectory) {
             context.reportError(
                 this,
-                MigrateUserData.FileDirectoryConflictException(
+                FileDirectoryConflictException(
                     sourceFile,
                     Directory.createInstanceUnsafe(destinationFile)
                 )
             )
-            return operationCompleted()
+            return
         }
 
-        if (handledEquivalentFileContent(destinationExists, context)) {
-            return operationCompleted()
-        }
-
-        // destination exists, does NOT match content, and is 0-length
-        // delete it and let the transfer occur again.
-        if (destinationExists && destinationFile.length() == 0L) {
-            // TODO: #13170 - extend this for when destinationFile is an exact subset of sourceFile
-            destinationExists = !destinationFile.delete()
-            Timber.w("(conflict) Deleted empty file in destination. Deletion result: %b", destinationExists)
+        when (FileUtil.isPrefix(destinationFile, sourceFile.file)) {
+            FileUtil.FilePrefix.EQUAL -> {
+                // Both files exist and are identical. Delete source + report size
+                context.execSafe(this) {
+                    val fileSize = sourceFile.file.length()
+                    deleteFile(sourceFile.file)
+                    context.reportProgress(fileSize)
+                }
+            }
+            FileUtil.FilePrefix.STRICT_PREFIX -> {
+                val deletionSucceeded = destinationFile.delete()
+                Timber.w("(conflict) Deleted empty file in destination. Deletion result: %b", deletionSucceeded)
```

Comment:
```suggestion
                Timber.w("(conflict) Deleted partial file in destination. Deletion result: %b", deletionSucceeded)
```

technically an empty file is a partial file isn't it? anyway now that you are checking for prefixes empty is definitely not correct, partial seems more correct. This is a nit though